tests/utils/git_test.py

```diff
@@ -87,176 +163,75 @@ def test_git_version_too_old(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = [
-            "  v1",
-            "  v2",
-        ]
-        assert git.tags("/src") == ["v1", "v2"]
-        run_subprocess.assert_called_with("git -C /src tag")
-
-    @mock.patch("esrally.utils.process.run_subprocess_with_output")
-    @mock.patch("esrally.utils.process.run_subprocess_with_logging")
-    def test_list_tags_no_tags_available(self, run_subprocess_with_logging, run_subprocess):
-        run_subprocess_with_logging.return_value = 0
-        run_subprocess.return_value = ""
-        assert git.tags("/src") == []
-        run_subprocess.assert_called_with("git -C /src tag")
+            git.checkout(TestGit.local_tmp_src_dir, branch=branch)
+        assert exc.value.args[0] == f"Could not checkout [{branch}]. Do you have uncommitted changes?"
+
+    def test_checkout_revision(self):
+        # Apple Git 'core.abbrev' defaults to return 7 char prefixes (09980cd)
+        # Linux defaults to return 8 char prefixes (09980cd5)
+        git.checkout_revision(TestGit.local_tmp_src_dir, revision="bd368741951c643f9eb1958072c316e493c15b96")
+        assert git.head_revision(TestGit.local_tmp_src_dir).startswith("bd36874")
+
+    def test_checkout_branch(self):
+        git.checkout_branch(TestGit.local_tmp_src_dir, remote=TestGit.remote_repo, branch=TestGit.remote_branch)
+        assert git.head_revision(TestGit.local_tmp_src_dir).startswith(TestGit.remote_branch_hash[0:7])
+
+    def test_head_revision(self):
+        # Apple Git 'core.abbrev' defaults to return 7 char prefixes (09980cd)
+        # Linux defaults to return 8 char prefixes (09980cd5)
+        git.checkout(TestGit.local_tmp_src_dir, branch="2.6.0")
+        assert git.head_revision(TestGit.local_tmp_src_dir).startswith("09980cd")
+
+    def test_pull_ts(self):
+        # results in commit 28474f4f097106ff3507be35958db0c3c8be0fc6
+        git.pull_ts(TestGit.local_tmp_src_dir, "2016-01-01T110000Z", remote=TestGit.remote_repo, branch=TestGit.remote_branch)
+        assert git.head_revision(TestGit.local_tmp_src_dir).startswith("28474f4")
+
+    def test_rebase(self, setup_teardown_rebase):
+        # fetch required first to get remote branch
+        git.fetch(TestGit.local_tmp_src_dir, remote=TestGit.remote_repo)
+        git.rebase(TestGit.local_tmp_src_dir, remote=TestGit.remote_repo, branch=TestGit.rebase_branch)
+        ops = " ".join(process.run_subprocess_with_output(f"git -C {TestGit.local_tmp_src_dir} reflog --since=10.seconds.ago"))
+        # reflog format differs from Apple Git to Linux
+        finished_pattern = re.compile(f"rebase (\\(finish\\)|finished).*returning to refs\\/heads\\/{TestGit.rebase_branch}")
+        started_pattern = re.compile(f"rebase:? (\\(start\\)*.|checkout).*{TestGit.remote_repo}\\/{TestGit.rebase_branch}")
+        assert re.search(started_pattern, ops)
+        assert re.search(finished_pattern, ops)
```

Comment:
This is brittle, not only because of the reflog format, but also because of the 10 seconds. We could match something that we do not want, and the machine could also hand for 10 seconds (especially in CI, say because a live migration happened or something).

What do you think about committing to the remote repo and then checking that the commit is here after the rebase?

Reply:
Nice suggestion, and agreed RE: brittle. Addressed in https://github.com/elastic/rally/pull/1602/commits/8470aee7f82a920dfefc7bae0023761a028445f8